### DISCUSSION TITLE
Using getCurrentUser in global auth middle causes No Firebase App error

### DISCUSSION THREAD

**Discussion by johnpuaoi** [vuejs/vuefire#1388] [Questions and Help] [answered]
Trying to follow the nuxt example and check for the current user in middleware but doing so causes this error: Firebase: No Firebase App '[DEFAULT]' has been created - call initializeApp() first (app/no-app).

Here is my auth.global.ts middleware.

```ts
import { getCurrentUser } from 'vuefire';
export default defineNuxtRouteMiddleware(async (to, from) => {
  const user = await getCurrentUser();
  // IF the user is going to the auth page and there is no user
  if (to.path === '/' && !user) {
    return;
  }

  if (!user) {
    return navigateTo({
      path: '/',
      query: {
        redirect: to.fullPath,
        authMode: 'login',
      },
    });
  }
});
```

I am using the GOOGLE_APPLICATION_CREDNETIALS env var with it being an object instead of a path to a serviceAccount json file.

Any help would be appreciated!

**Comment by johnpuaoi** [comment, marked as answer]
Realized there was an issue with my appcheck config and the import of getCurrentUser, corrected my app check config and changed middleware to below, issue is gone.

```ts
export default defineNuxtRouteMiddleware(async (to, from) => {
  const user = await getCurrentUser();

  // IF the user is going to the auth page and there is no auth user, let them go
  if (to.path === '/' && !user) {
    return;
  }

  // If no auth user and not going to auth page, redirect to auth page, with redirect query
  if (!user) {
    return navigateTo({
      path: '/',
      query: {
        redirect: to.fullPath,
        authMode: 'login',
      },
    });
  } else if (user && to.path === '/') {
    return navigateTo({
      path: '/dashboard',
    });
  }
});
```

**Reply by francois-launchbase** [reply]
Thanks for this answer! I spent at least two days trying to figure this out. I also had the unnecessary import statement that broke everything in ways that did not make sense 🤕
